types/index.d.ts

```diff
@@ -1482,6 +1482,26 @@ declare module 'mongoose' {
 
   type ProjectionFields<DocType> = { [Key in keyof Omit<LeanDocument<DocType>, '__v'>]?: any } & Record<string, any>;
 
+  type _QueryManyForModel<T, ResultDoc, TQueryHelpers = {}> =
+    QueryWithHelpers<Array<ResultDoc>, ResultDoc, TQueryHelpers, T>;
+  type _QueryOneForModel<T, ResultDoc, TQueryHelpers = {}> =
+    QueryWithHelpers<ResultDoc, ResultDoc, TQueryHelpers, T>;
+
+  /**
+   * Helper type for getting a find Query type for a given Model (return of Model.find)
+   */
+  type QueryForModel<T extends Model<any>> =
```

Comment:
That was added in #11650 -- there is an open question in the other branch on if you want me to pull it out. I find it to be a useful helper for getting the type of Query but it can be removed